Neuro/Minigames/Solvers/AvertCrashCourseSolver.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Neuro.Cursor;
+using System.Collections;
+using System.Linq;
+using Neuro.Utilities;
+using UnityEngine;
+
+namespace Neuro.Minigames.Solvers;
+
+[MinigameSolver(typeof(AirshipAuthGame))]
+public sealed class AvertCrashCourseSolver : IMinigameSolver<AirshipAuthGame>, IMinigameOpener
+{
+    public bool ShouldOpenConsole(Console console, PlayerTask task) => true;
+
+    public IEnumerator CompleteMinigame(AirshipAuthGame minigame)
+    {
+        do
+        {
+            yield return new WaitForSeconds(0.5f);
+            int codeNumber = minigame.system.TargetCode;
+            int[] code = codeNumber.ToString().PadLeft(5, '0').Select(c => c - '0').ToArray();
+            foreach (int number in code)
+            {
+                yield return InGameCursor.Instance.CoMoveTo(minigame.selectableButtons.At(number > 0 ? number - 1 : 9));
+                minigame.ClickNumber(number);
+                yield return new WaitForSeconds(0.25f);
+
+                if (codeNumber != minigame.system.TargetCode)
+                {
+                    yield return InGameCursor.Instance.CoMoveTo(minigame.selectableButtons.At(^2));
+                    minigame.ClearEntry();
+                    yield return new WaitForSeconds(0.5f);
+
+                    // We don't have a cleaner way to exit
+                    yield return CompleteMinigame(minigame);
+                    yield break;
```

Comment:
If the code resets while we're typing itt, we would need to basically execute a continue for the parent do-while loop, but since we're in a foreach we can't really do that cleanly and I don't want to use state booleans or anything like that. If you have any better ideas on how to implement this maybe we can try that? @krogenth

Reply:
I can take a look and see what I can come up with for this.

Reply:
For now what I've added works it just seems a bit scuffed

Reply:
I believe we could do something like the following:
1. instead of `yield break`, we can just `break;` instead inside the foreach
2. move all of the entering code inside the `if (codeNumber == minigame.system.TargetCode)` block

that should at least stop the recursion from being required

can push it up if you want to check it out

Reply:
Sounds good to me

Reply:
Changes pushed, feel free to verify before doing anything.

Reply:
Looks good, can you please copy this behaviour over to the comms solver? I'll merge this pull request afterwards.

Reply:
Done.